### PR TITLE
Adds a new anomaly core item to Illumination technology, the photon cannon.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request
This PR adds a new gunkit to the **Illumination Applications** tech node acquired via BEPIS techs. For years, this funny little tech node has been one of the lamer results from the BEPIS tech pool, only offering constructable unique hardhats and access to a micro RLD. Well, lets just say, badda bing

https://github.com/tgstation/tgstation/assets/41715314/bd869d7f-449a-4563-871a-b90f550e0bda

The photon cannon is craftable via a lathed gunkit obtained from the security lathe, which fires 4 shots per recharge. Each shot, on direct contact with a mob, acts like a severity 3 flash. The shot also distinctly slows down after traveling, so you can more effectively gauge where it's going to stop moving, as when it reaches it's travel distance, it will perform a weaker AOE severity 1 flash. Direct hits will effect silicons, while the AOE flash will not. Direct hits only deal 5 ***STAMINA* damage.

The Gunkit requires a flux anomaly core to be crafted, completing with the tesla gun in this respect.

## Why It's Good For The Game

There's been some discussion about how we've hoping to offer higher power features and options to the game at large using anomaly cores, and flux anomalies are currently only used for guns, which is rather unfortunate. The bigger issue, I'd suppose, is that every gun that flux anomalies can produce are considered lethal, with the event horizon gun and tesla gun. And honestly, by comparison, the tesla is not lethal enough to use to any real effect, or practical enough to bother science enough for the core.

The photon cannon, however, works as a rather effective crowd control weapon for breaching and incapacitating crewmates, making it ideal for wardens dealing with large tides of crewmates, officers breaking into bases, or ~~the captain trying to burn down your library dnd game, DOUGLAS~~

Overall, it's still a fairly low power weapon, and sits in a niche place somewhere between a dragNET, flashbang-loaded grenade launcher, the flash, and the ion-cannon,  with it's versatility coming from core rarity and requiring a BEPIS tech.
**I have since updated this PR so that this gun deals stamina damage instead of energy damage, and contact with the energy ball deals a short period of confusion.**

## Changelog

:cl:
add: The Illumination Applications technode now includes a gunkit for a new gun, the Photon Cannon.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile/energy/photon.dm [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	RegisterSignals(src, list(COMSIG_MOVABLE_CROSS, COMSIG_MOVABLE_CROSS_OVER), PROC_REF(blast_touched))
+	RegisterSignal(src, COMSIG_ATOM_ENTERED, PROC_REF(scorch_earth))
+	set_light_on(TRUE)
+
+/**
+ * Handle side effects for the phonon bolt.
+ * behaves like a higher power direct flash if hit, and sparks silicons like they're getting microwaved.
+ */
+/obj/projectile/energy/photon/proc/blast_touched(datum/source, atom/flashed)
+	if(isliving(flashed))
```
Comment (MrMelbert) [inline code/modules/projectiles/projectile/energy/photon.dm:29]: `SIGNAL_HANDLER`

File: code/modules/projectiles/projectile/energy/photon.dm [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		var/mob/living/flashed_creature = flashed
+		flashed_creature.flash_act(intensity = 3, affect_silicon = TRUE, length = 6)
+	if(issilicon(flashed))
+		do_sparks(rand(1, 4), FALSE, src)
+
+/**
+ * When traveling to a new turf, throws a probability to generate a hotspot across it's path.
+ */
+/obj/projectile/energy/photon/proc/scorch_earth(turf/open/floor/source, atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	if(prob(40))
```
Comment (MrMelbert) [inline code/modules/projectiles/projectile/energy/photon.dm:39]: `SIGNAL_HANDLER`

File: code/modules/projectiles/projectile/energy/photon.dm [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+/obj/projectile/energy/photon/Range()
+	. = ..()
+	pixel_speed_multiplier *= MULTIPLY_PIXELSPEED
+
+/obj/projectile/energy/photon/on_range()
+	do_sparks(rand(4, 9), FALSE, src)
+	playsound(loc, 'sound/weapons/solarflare.ogg', 100, TRUE, 8, 0.9)
+	for(var/mob/flashed_mob in viewers(5, loc))
+		if(!isliving(flashed_mob))
+			continue
```
Comment (MrMelbert) [inline code/modules/projectiles/projectile/energy/photon.dm:51]:
```suggestion
	for(var/mob/living/flashed_mob in viewers(5, loc))
```

File: code/modules/projectiles/projectile/energy/photon.dm [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		var/mob/living/flashed_living_mob = flashed_mob
+		flashed_living_mob.flash_act()
```
Comment (MrMelbert) [inline code/modules/projectiles/projectile/energy/photon.dm:53]:
```suggestion
		flashed_mob.flash_act()
```

**Comment by ATH1909** [comment]
the 5 damage from a direct hit makes this break asimov, which is annoying

**Comment by necromanceranne** [comment]
Since it does actually do lethal damage, it cannot be considered nonlethal, but less-than-lethal.

It probably should just deal stamina damage since it is just a ranged flash.

**Comment by moocowswag** [comment]
As long as flux anomaly core isnt used to bring back tasers im fine with a nonlethal gun

**Comment by dopamiin0** [comment]
> As long as flux anomaly core isnt used to bring back tasers im fine with a less than lethal gun
> 
> > the 5 damage from a direct hit makes this break asimov, which seems like an unnecessary restriction
> 
> Flashing does eye damage too and multiple can blind someone but the world isnt ready for that implication

as far as im aware organ damage generally isnt considered asimov breaking, as weird as that sounds. otherwise basically any application of medicine would be harm to humans. its dumb and semantic but i do think it would be actually more usable if it didnt deal the 5 damage

**Comment by YuliceTheDruid** [comment]
Honestly while I appreciate the anomaly core resurgence as more content is good for the game this honestly doesn't seem that great for an anomaly core weapon. Grenade Launcher from armory with flash bangs will do this waaaaay better. Could definitely stand to make it beefier if you want to invest cores into it.

**Comment by Vect0r2** [comment]
> Honestly while I appreciate the anomaly core resurgence as more content is good for the game this honestly doesn't seem that great for an anomaly core weapon. Grenade Launcher from armory with flash bangs will do this waaaaay better. Could definitely stand to make it beefier if you want to invest cores into it.

Nah I kinda like it for what it is: it's a beepis tech and it is strange and funny like a beepis tech should be.

**Comment by zxaber** [comment]
A Sev 3 flash defeats all individual flash protection, right? You'd need a weld helmet on top of sunglasses (or goggles if you're a moth) to meet that level of bright.

So it's basically a guaranteed knockdown on most humanoids, including other sec. The most likely to be immune3 would be an engineer in a modsuit after they raided a sec locker.

**Comment by Exester509** [comment]
> Honestly while I appreciate the anomaly core resurgence as more content is good for the game this honestly doesn't seem that great for an anomaly core weapon. Grenade Launcher from armory with flash bangs will do this waaaaay better. Could definitely stand to make it beefier if you want to invest cores into it.

Considering this is pitched as an alternative to the tesla cannon and bepis tech, this is about as strong as you'd expect after hearing both of those things.

**Comment by EuclidianGeometer** [comment]
> Considering this is already locked behind bepis and isn't too strong anyway, I think it would be better if it had no anomaly core requirement.

bepis tech items are broadly 'goofy useful' things that are low powered but helpful, or high powered but with huge downsides. I think having an actually good item that comes from it that has fairly small if any downsides to the users should have an anomaly core requirement.

**Comment by Loafin34** [comment]
could you delay the explosion so that you have a temporary floodlight?

**Comment by necromanceranne** [comment]
Unironically, a gun that just shoots a big ball of temporary light would be really neat. I don't even care about a new weapon I'd love a utility tool like that.

**Comment by LSommers** [comment]
Even better: ball of light that you can detonate like a flashbang by clicking it again or something.

Also this thing should straight up dust a nightmare on a direct hit, it would be funny.

**Comment by EricZilla** [comment]
I feel like this is good to be added, ita going to be rare anyway and if the values need tweeking later it's fine.
I kinda think it should have a really slow self charge to feel a little more anomalous, but idk

**Comment by Ghommie** [comment]
A few suggestions: the photon projectile should live another extra few seconds or so once it reaches its destination, so that it can keep the area illuminated longer, and perhaps it should also confuse whoever is hit and successfully flashed by it.

**Comment by CCC23** [comment]
flashbang

**Pull request merged**